AG-16239 - BarSeries performance improvements Pt 3e

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-16239

alantreadway committed:
AG-16239 - Micro-optimisations for bar-series incremental data updates.

alantreadway committed:
AG-16239 - Micro-optimisations for bar-series incremental data updates.

alantreadway committed:
AG-16239 - Micro-optimisations for bar-series incremental data updates.

alantreadway committed:
AG-16239 - Micro-optimisations for bar-series incremental data updates.

alantreadway committed:
AG-16239 - Add DataChangeDescription to ProcessedData.

alantreadway committed:
AG-16239 - Improve aggregation.ts ternary usage.

alantreadway committed:
AG-16239 - Add 'roll' action to examples.

alantreadway committed:
AG-16239 - Fix ExtendedPath2D.clear()

alantreadway committed:
AG-16239 - Cleanup aggregation.ts ternary use.

alantreadway committed:
AG-16239 - Update snapshots.

lsjroberts commented on packages/ag-charts-community/src/module/__snapshots__/optionsModule.test.ts.snap:
```diff
@@ -5266,9 +5266,7 @@ exports[`ChartOptions #prepareOptions for COLUMN_CHART_WITH_NEGATIVE_VALUES_EXAM
       "type": "bar",
       "visible": true,
       "xKey": "month",
-      "xKeyAxis": "x",
```
Any idea why these snapshots updated? (This is the `column-with-negative-values` example).

alantreadway replied:
Oddly this is failing for me on `latest` - not sure why CI hasn't failed

alantreadway commented on packages/ag-charts-community/src/chart/series/cartesian/cartesianSeries.ts:
```diff
@@ -562,7 +566,17 @@ export abstract class CartesianSeries<
+        // Skip selection update if the same datum is still highlighted
+        const currentDatumIndex = item.datumIndex;
+        if (currentDatumIndex === this.lastHighlightedDatumIndex) {
+            return true; // Still highlighted, but skip expensive selection updates
```
Fix incoming.

lsjroberts reviewed (approved):
LGTM

alantreadway committed:
AG-16239 - Revert highlight selection cache optimization

Remove the highlight datum index caching as it prevents style/drawing-mode
updates from being applied when the same datum remains highlighted.
